doc/api/stream.md

```diff
@@ -1590,6 +1596,15 @@ async function run() {
 run().catch(console.error);
 ```
 
+On completion or error `stream.pipeline()` will call `stream.destroy(err)` on
+all provided stream except for `Readable` streams which have emitted `'end'`
+or `'close'` and `Writable` streams which have emitted `'finish'` or `'close'`.
+Unless `autoDestroy` is enabled these streams will not be automatically cleaned
+up.
+
+If any `Writable` or `Readable` stream emits `'close'` without fully flushing
+or draining, `stream.pipeline()` will error with `ERR_STREAM_PREMATURE_CLOSE`.
```

Comment:
This is a bit vague on purpose. Basically, the only thing that can be assumed is `ERR_STREAM_PREMATURE_CLOSE` will be emitted if something happened which caused data to be lost.